python/lsst/ctrl/bps/bps_core.py

```diff
@@ -118,43 +118,26 @@ def pretty_dataset_label(orig_name):
     return new_name
 
 
-def save_single_qgnode(qgnode, out_filename):
-    """Save single quantum to file
+def save_qg_subgraph(node_ids, qgraph, out_filename):
+    """Save subgraph to file
 
     Parameters
     ----------
-    qgnode : QuantumGraph Node
-        Single quantum to save
+    node_ids : `lsst.pipe.base.graph.quantumNode.NodeId` or
+               iterable of `lsst.pipe.base.graph.quantumNode.NodeId`
+        NodeIds for Quanta inside given qgraph to save
     out_filename : `str`
         Name of the output file
     """
-    os.makedirs(os.path.dirname(out_filename), exist_ok=True)
-    qgraph2 = QuantumGraph()
-    qgraph2.append(qgnode)
-    with open(out_filename, "wb") as pickle_file:
-        pickle.dump(qgraph2, pickle_file)
-
-
-def count_quantum(qgraph):
-    """Count Quantum in the QuantumGraph
 
-    Parameters
-    ----------
-    qgraph : `QuantumGraph`
-        QuantumGraph that needs counting
+    # create subgraph
+    qnodes = [qgraph.getQuantumNodeByNodeId(id_) for id_ in iterable(node_ids)]
+    subgraph = qgraph.subset(qnodes)
```

Comment:
consider:
```python
subgraph = qgraph.subset(qgraph.getQuantumNodeByNodeId(id_) for id_ in iterable(node_ids))
```
Subset already accepts a single Node or an iterable of Nodes, so no need to use iterable here so if you kept reference to the nodes it would just be:
```python
subgraph = qgraph.subset(nodes)
```

Reply:
Went with the reference to the QuantumNodes instead of the NodeId, so the latter suggestion was implemented here.